### DISCUSSION TITLE
A way to undelete episodes

### DISCUSSION THREAD

**Discussion by capturts** [jeff-hughes/shellcaster#47] [Ideas]
Sometimes I've gone to remove an episode from the list and either pressed "r" on the wrong one or changed my mind afterwards. Is there a way to show removed items. I'm guessing it's deleted from the database, so I assume not. But some sort of re-sync that would show deleted ones (maybe with a "-" or something to show they were previously removed from the list and a "u" press to undelete. Could that be a thing?

**Comment by jeff-hughes** [comment]
Look, this program is not made for people who "make mistakes". It's for people who are perfect, all the time, every time...

Haha okay, yeah that's a good idea. When it comes to episodes, the removed ones aren't actually deleted from the database -- otherwise, when syncing the podcast again, it would just say "oh look, here's a new episode we haven't seen yet" and re-insert it. So as it stands right now, you could open the database and just change the "hidden" flag from 1 to 0, and it will appear again. That's obviously not a great solution, but it is an option that exists _today_ to undelete an episode if you run into the situation. [DB Browser](https://sqlitebrowser.org) is a great option if you're looking for a simple GUI for viewing/editing SQLite databases.

I've been toying around with a better option for "editing" podcasts in general. There are a couple of open issues (#32 and #44) that would benefit from a more full-featured option for editing an existing podcast. I'm thinking of a separate window that pops up with the podcast details, so you're able to add in a username/password for private feeds, to edit the URL if the podcast moves, and so on. So that might also be a good place for a list of all episodes, including the previously hidden ones, which you could check/uncheck.

With that said, that will take some work to create, and my time has been pretty limited lately, so this isn't something that will be showing up next week by any means. But it's a great suggestion and I'll definitely add it to the list 🙂
